// jotform/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const form_1 = __importDefault(require("./form"));
const user_1 = __importDefault(require("./user"));
const client_1 = __importDefault(require("./client"));
const report_1 = __importDefault(require("./report"));
const submission_1 = __importDefault(require("./submission"));
const folder_1 = __importDefault(require("./folder"));
const system_1 = __importDefault(require("./system"));
class Jotform {
    constructor(apiKey, options) {
        const instance = new client_1.default(apiKey, options);
        this.user = new user_1.default(instance);
        this.form = new form_1.default(instance);
        this.report = new report_1.default(instance);
        this.submission = new submission_1.default(instance);
        this.folder = new folder_1.default(instance);
        this.system = new system_1.default(instance);
    }
}
exports.default = Jotform;
